torchgeo/datasets/eurosat.py

```diff
@@ -0,0 +1,198 @@
+    )
+    class_counts = {
+        "AnnualCrop": 3000,
+        "Forest": 3000,
+        "HerbaceousVegetation": 3000,
+        "Highway": 2500,
+        "Industrial": 2500,
+        "Pasture": 2000,
+        "PermanentCrop": 2500,
+        "Residential": 3000,
+        "River": 2500,
+        "SeaLake": 3000,
+    }
+    class_name_to_label_idx = {
+        "AnnualCrop": 0,
+        "Forest": 1,
+        "HerbaceousVegetation": 2,
+        "Highway": 3,
+        "Industrial": 4,
+        "Pasture": 5,
+        "PermanentCrop": 6,
+        "Residential": 7,
+        "River": 8,
+        "SeaLake": 9,
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new EuroSAT dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.fns = []
+        self.labels = []
+        for class_name, class_count in self.class_counts.items():
+            for i in range(1, class_count + 1):
+                self.fns.append(
+                    os.path.join(
+                        self.root, self.base_dir, class_name, f"{class_name}_{i}.tif"
+                    )
+                )
+                self.labels.append(self.class_name_to_label_idx[class_name])
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        sample: Dict[str, Tensor] = {
+            "image": self._load_image(index),
+            "label": torch.tensor(self.labels[index]),  # type: ignore[attr-defined]
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.labels)
+
+    @lru_cache()
```

Comment:
Does this need to be cached? I don't think the cache will be hit very often since it only contains the most recent 128 images and the total dataset size is 27K. So it may actually slow things down due to overhead.

Reply:
I don't think it matters